packages/core/database/lib/query/query-builder.js

```diff
@@ -301,10 +301,8 @@ const createQueryBuilder = (uid, db) => {
           break;
         }
         case 'count': {
-          const dbColumnName =
-            state.count === '*' ? '*' : this.aliasColumn(helpers.toColumnName(meta, state.count));
-
-          qb.count({ count: dbColumnName });
+          const dbColumnName = this.aliasColumn(helpers.toColumnName(meta, 'id'));
```

Comment:
This removes a feature for counting on a specific column. This is a breaking change.

Reply:
@alexandrebodin ok, how do we exactly count by a specific column?

Reply:
I was not aware that counting on a specific column existed either, sorting yes obviously but counting I believe should be based on the ID.

Unless this is an undocumented feature?

Reply:
Pardon the ping but I ran into this issue today with a total count that includes a deeply nested self reference which might actually be caused by this column count feature mentioned here. The issue may not be with the actual counting logic but rather the column counting logic more specifically 

schema is roughly: 
```
product: {
    make: { // relation
        models // relation
    },
    model: { // relation
        years, make // two relations
    },
    year: { // relation
        make, model // two relations
    },
}
```

In my scenario it seems like the total count should be 1 but is actually 12 (max for the request)  I assume this is coming from size * count but pageCount should not be 12. The data returned is correct but the pagination count appears to be going multiple layers deep looking for matches (and possibly populating all references all the way down further than my request is intending).

pagination response from request looking for make_name and model_name matches on one specific slug each 
{
	"pagination": {
		"page": 1,
		"pageSize": 12,
		"pageCount": 12,
		"total": 144
	}
}

Reply:
Counting on a column is a SQL feature. It excludes null values so you can count the real number of rows that actually have a value for a column. 

It is also useful when using distinct counts as it avoids having to use group by statements. 

@Bassel17 you can update the code slightly to still accept the `state.count` if defined and use id if not so we can keep that feature.

Reply:
@alexandrebodin yes got it, I pushed a fix for it